### PR TITLE
[Doc] API documentation updates.

### PR DESCRIPTION
**Pull request by dan-zheng**
Assorted documentation updates.

### CODE REVIEW

File: stdlib/public/Python/NumpyConversion.swift [hunk 1]
```diff
@@ -93,6 +93,13 @@ extension Double : NumpyScalarCompatible {
 
 extension Array : ConvertibleFromNumpyArray
   where Element : NumpyScalarCompatible {
+  /// Creates an `Array` with the same shape and scalars as the specified
+  /// `numpy.ndarray` instance.
+  ///
+  /// - Parameter numpyArray: The `numpy.ndarray` instance to convert.
+  /// - Precondition: The `numpy` Python package must be installed.
+  /// - Precondition: `numpyArray` must be 1-D and have a compatible scalar
```
Comment (mhong) [inline stdlib/public/Python/NumpyConversion.swift]: do you want to more explicitly describe when a nil value will be returned? Same for other APIs below.

`Precondition` usually describes cases where the code will lead to undefined behavior (e.g. crash) when the condition is violated.

**Review by mhong — APPROVED** [review]
nice!

File: stdlib/public/TensorFlow/NumpyConversion.swift [hunk 2]
```diff
@@ -87,6 +93,12 @@ extension ShapedArray : ConvertibleFromNumpyArray
 
 extension Tensor : ConvertibleFromNumpyArray
   where Scalar : NumpyScalarCompatible {
+  /// Creates a tensor with the same shape and scalars as the specified
+  /// `numpy.ndarray` instance.
+  ///
+  /// - Parameter numpyArray: The `numpy.ndarray` instance to convert.
+  /// - Precondition: The `numpy` Python package must be installed.
+  /// - Precondition: `numpyArray` must have a compatible scalar datatype.
```
Comment (rxwei) [inline stdlib/public/TensorFlow/NumpyConversion.swift]: "datatype" isn't a thing in either Numpy or Swift. `dtype` (backquoted) is probably more appropriate.

```suggestion
  /// - Precondition: `numpyArray` must have a compatible scalar datatype.
```

File: stdlib/public/Python/NumpyConversion.swift [hunk 3]
```diff
@@ -93,6 +93,13 @@ extension Double : NumpyScalarCompatible {
...
+  ///   datatype.
```
Comment (rxwei) [inline stdlib/public/Python/NumpyConversion.swift]:
```suggestion
  ///   `dtype`.
```

**Review by rxwei — APPROVED** [review]

File: stdlib/public/Python/NumpyConversion.swift (hunk 1 above)
Comment (rxwei) [reply stdlib/public/Python/NumpyConversion.swift]: +1. Precondition is definitely not the right term for describing `nil`. We should look at how common failable initializers are documented in the standard library.

File: stdlib/public/TensorFlow/NumpyConversion.swift (hunk 2 above)
Comment (dan-zheng) [reply stdlib/public/TensorFlow/NumpyConversion.swift]: Thanks! Done in https://github.com/apple/swift/pull/23069/commits/e7920e443eeb844ac662c91069d9c9025a3c2439.

File: stdlib/public/Python/NumpyConversion.swift (hunk 3 above)
Comment (dan-zheng) [reply stdlib/public/Python/NumpyConversion.swift]: Done in https://github.com/apple/swift/pull/23069/commits/e7920e443eeb844ac662c91069d9c9025a3c2439.

File: stdlib/public/Python/NumpyConversion.swift (hunk 1 above)
Comment (dan-zheng) [reply stdlib/public/Python/NumpyConversion.swift]: Thanks for great catch!

A `Returns:` comment is typically used to describe failable initializers and when they return `nil`.

Updated in https://github.com/apple/swift/pull/23069/commits/e7920e443eeb844ac662c91069d9c9025a3c2439, following [this stdlib documentation](https://github.com/apple/swift/blob/948655e850e0a4dd8c0f6d5a86f20f2c462c7303/stdlib/public/core/CString.swift#L137).

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Review by mhong — APPROVED** [review]

**Pull request merged**